contracts/EnigmaSimulation.sol

```diff
@@ -170,16 +210,32 @@ contract EnigmaSimulation is EnigmaStorage, EnigmaEvents, Getters, Ownable {
     * worker. This should be called by every worker (and the principal)
     * node in order to receive tasks.
     *
+    * @param _stakingAddress The operating address
     * @param _signer The signer address, derived from the enclave public key
     * @param _report The RLP encoded report returned by the IAS
     * @param _signature Signature
     */
-    function register(address _signer, bytes memory _report, bytes memory _signature)
+    function register(address _stakingAddress, address _signer, bytes memory _report, bytes memory _signature)
     public
     isUpdatedEnigmaContract
+    workerUnregistered(msg.sender)
```

Comment:
Same comment: 
Why do you check that the worker is not registered? Isn't the `isUniqueSigningKey` require is enough?
If it is necessary, comparing only to the Registered status is not enough, other statuses should be checked as well...

Reply:
Same as above!